llap-tez/src/java/org/apache/hadoop/hive/llap/tezplugins/LlapTaskSchedulerService.java

```diff
@@ -1386,8 +1386,9 @@ boolean isRequestedHostPresent(TaskInfo request) {
    */
   private SelectHostResult selectHost(TaskInfo request) {
     String[] requestedHosts = request.requestedHosts;
-    String requestedHostsDebugStr = Arrays.toString(requestedHosts);
+    String requestedHostsDebugStr = null;
     if (LOG.isDebugEnabled()) {
+      requestedHostsDebugStr = Arrays.toString(requestedHosts);
```

Comment:
Lets add a comment here mentioning that requestedHostsDebugStr should be properly assigned on the FIRST isDebugEnabled block.